_posts/2019-01-05-its-time-to-use-spm.markdown

```diff
@@ -0,0 +1,158 @@
+projects. It'll take a bit of sacrifice and a little bit of community spirit to fix some holes probably but **in my
+opinion, it's time for teams to start adopting SPM for their 3rd party dev tools**.
+
+**TDLR:** You should be using SPM for 3rd party dev tools like: [SwiftLint][sl], [SwiftFormat][sf], [Danger][ds],
+[Sourcery][srcy], [SwiftGen][sg] and [Git Hook management][kom].
+
+This post covers: What made it feasible to use SPM now? What are the downsides of the status quo? Why use SPM at
+all? What are the downsides to using SPM?
+
+<!-- more -->
+
+### What changed to make SPM usable?
+
+From my perspective, David Hart's [addition of `swift run`][swift-run] to SPM which shipped with Swift 4.0 is what
+pushed the project over the finish line to being useful for iOS developers. `swift run` is contextually the same as
+`bundle exec` in that it will run a locally bundled version of your executable.
+
+This means you can run `run swift swiftlint` and reliably get the same results as your CI and fellow developers.
+
+Second, all of the big third party tools support SPM already. So, you probably don't need to send any upstream PRs.
+
+### What are the downsides of the status quo
+
+**Using Homebrew**
+
+Right now, a lot of folks use [homebrew][] to manage these types of dependencies. Homebrew is useful for rarely
+updated tools (like unix-y CLI apps) but it does not handle having different versions of tool available. This is a
+totally reasonable call from Homebrew's perspective but it makes homebrew a bad choice for your **project**
+dependencies - because it only installs things globally.
+
+This means a developer (or your CI) would get the most recent version of that tool when they last installed the
+tool. This isn't a problem for many projects (for example, check out their [most installed][brew-top90] formulas to
+see that it's lot of system libraries, languages and global tools like `node`, `git` and `python`)
+
+**Using CocoaPods**
+
+You can hijack CocoaPod's dependency resolver, and locking system mixed with consistent dependency paths to handle
+your tools. This is better than using Homebrew, because everyone has the save version - and so you could reliably
+run SwiftLint via `./Pods/SwiftLint/swiftlint`.
+
+This is a great hack, and CocoaPods is smart here - because these dependencies don't ship any code for your app for
+compiling it won't set up a library or framework for you. You can even use CocoaPods to set up [a build
+phase][script_phase] for you too (I have feelings on this but we'll get to those later.)
+
+I don't really have much of a "you shouldn't do this" for using hacking CocoaPods for your tools, outside of SPM
+it's probably the right way to do it.
+
+### Why Use SwiftPM?
+
+1. SPM works
+1. SPM can lock your dependencies correctly. :tada:
+1. The major ecosystem tools already support it, so you don't need to do any extra work
```

Comment:
>The major ecosystem tools

I definitely understood the meaning and intent but that phrasing read a bit clunky to me. Maybe something like this would be better?

>The primary tools used in our ecosystem